SimplenoteTests/AuthenticationValidatorTests.swift

```diff
@@ -0,0 +1,63 @@
+import XCTest
+import XCTest
+@testable import Simplenote
+
+
+// MARK: - AuthenticationValidator Tests
+//
+class AuthenticationValidatorTests: XCTestCase {
+
+    /// Testing Validator
+    ///
+    let validator = AuthenticationValidator(style: .signup)
+
+
+    /// Verifies that `performUsernameValidation` returns `true` when the input string is valid
+    ///
+    func testPerformUsernameValidationReturnsTrueWheneverInputEmailIsValid() {
+        let results = [
+            validator.performUsernameValidation(username: "j@j.com"),
+            validator.performUsernameValidation(username: "something@simplenote.blog"),
+            validator.performUsernameValidation(username: "something@simplenote.blog"),
+            validator.performUsernameValidation(username: "something@simplenote.blog.ar")
+        ]
+
+        for result in results {
+            XCTAssert(result == .success)
```

Comment:
Nitpick: If you use `XCTAssertEqual` you can have more meaningful error messages. Something like:
`XCTAssertEqual failed: ("") is not equal to ("Password cannot match email")`
instead of
`XCTAssertTrue failed`

Reply:
Since now we've got `Equatable` this works ferpectly. Switching to XCTAssertEqual. **Thank you!!!!**